### ISSUE TITLE
Sample DAO follow-up grant

### ISSUE THREAD

**Issue by Itahand** [onflow/developer-grants#170]
# Expanding the Sample DAO on Flow

# Grant category

- Open source maintenance
- Educational material

# Description

This a follow-up grant from #97, since almost all milestones there are done at this moment. Two more milestones have surfaced and then I would like to ask for a monthly grant to keep maintaining the DAO, keep it bug free and up to date. 

The idea is to expand further on the sample code for most basic building blocks that are used for DAOs. I would like to include all the different ways we can explore an initial public sale, an initial dex sale, an allow-list sale and the DAO governance style, from the simplest to the more complicated structures, and even at different levels of decentralization. Users should be able to fork the project and have access to all of these different approaches to governance from the get-go. I will include documentation for each and every part of the open-source project, including the test suites, plus step-by-step guide articles and flow diagrams. 

This work will also include high quality level of designs and front-end work, similar to Touchstone from Emerald City. 

## Problem statement

The quality of the current repository is not high enough to represent the Flow ecosystem. Moreover, the current solution offer very limited and basic examples for the ICO, IDO and DAO smart contracts. in addition to this, the front-end work is clumsy and the user flow is hard to follow. 

- Target audience

    Flow Developers who want to build a DAO on Flow.
    Anyone who is new to the Flow ecosystem and curious to learn.

- Evidence for the need

Forking the current repository doesn't lead to an easy user experience. There's no real flow on the app, and all the pieces of the repository(Allow-list, ICO and IDO, and DAO) don't seem to work well together.

## Proposed solution

Diagrams: https://www.figma.com/file/MerF68gXa7OoNLlkRB2vYP/Flow-DAO-Open-Source?node-id=0%3A1&t=PKBolsXjUvdEDLYd-1

The proposal is: a complete and smooth user experience, with real designs and a functional and comprehensible front-end work. This app won't be a single page; instead it'll be multiple pages including but not limited to: Home page, contract generator, user guide, ICO dashboard/admin, IDO dashboard/admin and DAO dashboard/admin. 

The repository will include code example for all of the smart contracts, but will also provide extra smart contracts and transactions to allow users to deploy new tokens, ICOs and IDOs, and DAOs(with the ability to choose between different governance styles) from the front-end(the contract generator page of the website). 

## Impact

- In what ways does this benefit the broader Flow developer ecosystem?

Developers will have access to clear documentation and code samples on how to build a DAO using Cadence. As a result, it will help to onboard more developers on Flow. 

Non-developer users will also have an easy tool to launch their own contracts with guides on how to use them.  

# Milestones and funding

> Note: Please consider adoption and/or maintenance milestones at the end of your project.

| Milestone | Deliverables   | Timeline | Risks                   | USD proposal |
| --------- | -------------- | -------- | ----------------------- | -------------- |
| 1 - ICO examples   | Smart contracts + front-end  | 4-6 weeks  | - | $5,000           |
| 2 - IDO example with Increment integration | Smart contract + front-end | 2-4 weeks | - | $4,000 |
| 3 - DAO examples | Smart contracts + front-end  | 4-6 weeks | - | $8,000 |
| 4 - Adoption   | Creating tutorials (articles/documentation) for projects to integrate the project's smart contracts | 4 weeks  | - | $1,500           |
| 5 - Maintenance   | This is to maintain the contract in a proper manner given frequent Cadence updates | monthly  | - | $1,000           |

Total funding proposed: <total USD amount>
  
# Team

| Name | Role                | Bio | Contact         |
| ---- | ------------------- | --- | --------------- |
| Noah Naizir | Full-Stack Cadence Developer | Flow enthusiast | noah.naizir@gmail.com |

**Comment by franklywatson** [comment]
Hi @Itahand - thanks for submitting this grant proposal. We are presently making changes to the grant program and application process and as a result have not been able to review this. Once the grant program submissions are opened up again we will follow up and let you know next steps. Thanks for your patience.

**Comment by Aliserag** [comment]
We appreciate your proposal. After careful review we've decided not to proceed at this time. Please stay connected to our Twitter and working groups to stay on top of the latest developments in the ecosystem and new grant opportunities.
